### PR TITLE
WI #1383 Add UseExtractCopy notification

### PR DESCRIPTION
**Pull request by mayanje**

### CODE REVIEW

File: TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs [hunk 1]
```diff
@@ -91,6 +92,39 @@ protected virtual void OnDidReceiveNodeRefresh(NodeRefreshParams parameter)
...
+            catch (Exception e)
+            {
+                RemoteConsole.Error(String.Format("Error while handling notification {0} : {1}", notificationType.Method, e.Message));
+            }
+        }
+
+        protected virtual void OnDidReceiveExtractUseCopies(ExtractUseCopiesParams parameter)
+        {
+            var docContext = GetDocumentContextFromStringUri(parameter.textDocument.uri, false);
+            if (docContext.FileCompiler != null && docContext.FileCompiler.CompilationResultsForProgram != null &&
```
Comment (smedilol) [inline TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs]: You can use null propagration:
docContext.FileCompiler?.CompilationResultsForProgram?CopyTextNamesVariations != null

File: TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs [hunk 2]
```diff
@@ -91,6 +92,39 @@ protected virtual void OnDidReceiveNodeRefresh(NodeRefreshParams parameter)
...
+                docContext.FileCompiler.CompilationResultsForProgram.CopyTextNamesVariations != null)
+            {
+                var copiesName = docContext.FileCompiler.CompilationResultsForProgram.CopyTextNamesVariations.Select(cp => cp.TextNameWithSuffix).Distinct();
+                var missingCopiesParam = new MissingCopiesParams();
+                missingCopiesParam.textDocument = parameter.textDocument;
+                missingCopiesParam.Copies = new List<string>();
```
Comment (fm-117) [inline TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs]: You can use ToList() missingCopiesParam.Copies = docContext.FileCompiler.CompilationResultsForProgram.CopyTextNamesVariations.Select(cp => cp.TextNameWithSuffix).Distinct().ToList();

File: TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs (hunk 1 above)
Comment (mayanje) [reply TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs]: ok I wil do

File: TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs (hunk 2 above)
Comment (mayanje) [reply TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs]: yes

**Pull request merged**